File: kata-front-end/src/main/web/src/app/model/player.ts

export class Player {

    private _name: string ;
    private _gameScore: string;
    private _tieBreakScore: number;
    private _setScores: Map<number,number>;
    private _globalGameScore: string;


    constructor() {
        this.setScores = new Map<number,number>();
        this.setScores.set(1, null);
        this.setScores.set(2, null);
        this.setScores.set(3, null);
        this.setScores.set(4, null);
        this.setScores.set(5, null);
    }

    get name(): string {
        return this._name;
    }

    set name(value: string) {
        this._name = value;
    }

    get gameScore(): string {
        return this._gameScore;
    }

    set gameScore(value: string) {
        this._gameScore = value;
    }


    get setScores(): Map<number,number> {
        return this._setScores;
    }

    set setScores(value: Map<number,number>) {
        this._setScores = value;
    }


    get tieBreakScore(): number {
        return this._tieBreakScore;
    }

    set tieBreakScore(value: number) {
        this._tieBreakScore = value;
    }


    get globalGameScore(): string {
        return this._globalGameScore;
    }

    set globalGameScore(value: string) {
        this._globalGameScore = value;
    }
}

File: kata-back-end/src/main/java/com/sg/cib/tennis/kata/model/Player.java
package com.sg.cib.tennis.kata.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class Player {
    /**
     * Nom du joueur (voire de l'équpe).
     */
    private String name;

    /**
     * Score du joueur dans le jeu en cours.
     * A condition que ce dernier ne soit pas un jeu décisif.
     */
    private GameScores gameScore;

    /**
     * Score du joueur dans un jeu décisif.
     */
    private Integer tieBreakScore;

    /**
     * Liste contenant les numéros des sets gagnés par le joueur.
     */
    private List<Integer> wonSets;

    /**
     * Map des nombres de jeux gagnés par le joueur pour chacun des set du match.
     * La clée est donc le numéro du set.
     */
    private Map<Integer, Integer> setScores;

    public Player() {
    }

    public Player(String name, Integer nbOfSetsTowin) {
        this.name = name;
        this.gameScore = GameScores.ZERO;
        this.setScores = new HashMap<>();
        if(nbOfSetsTowin != null) {
            if (nbOfSetsTowin >= 1) {
                setScores.put(1, 0);
            }
            if (nbOfSetsTowin >= 2) {
                setScores.put(2, 0);
                setScores.put(3, 0);
            }
            if (nbOfSetsTowin == 3) {
                setScores.put(4, 0);
                setScores.put(5, 0);
            }
        }
        this.wonSets = new ArrayList<>();
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @JsonSerialize(using = GameStoresSerializer.class)
    public GameScores getGameScore() {
        return gameScore;
    }

    public void setGameScore(GameScores gameScore) {
        this.gameScore = gameScore;
    }

    public Map<Integer, Integer> getSetScores() {
        return setScores;
    }

    public void setSetScores(Map<Integer, Integer> setScores) {
        this.setScores = setScores;
    }

    public Integer getTieBreakScore() {
        return tieBreakScore;
    }

    public void setTieBreakScore(Integer tieBreakScore) {
        this.tieBreakScore = tieBreakScore;
    }

    public List<Integer> getWonSets() {
        return wonSets;
    }

    public void setWonSets(List<Integer> wonSets) {
        this.wonSets = wonSets;
    }
}
